server/libs/immich-config/src/immich-config.service.ts

```diff
@@ -21,6 +21,10 @@ const defaults: SystemConfig = Object.freeze({
     buttonText: 'Login with OAuth',
     autoRegister: true,
   },
+
+  storageTemplate: {
+    template: '{{y}}/{{y}}-{{MM}}-{{dd}}/{{filename}}.{{ext}}',
```

Comment:
`{{userId}}` is not included in the stored template, I don't think `{{.ext}}` should be either. 

The template string should be made up of things that user can change and we should add required things 

When the template is built we should always prepend the userId and always append ext to the final result.